lib/site_settings/validations.rb

```diff
@@ -55,6 +55,11 @@ def validate_enable_s3_uploads(new_val)
     validate_error :s3_upload_bucket_is_required if new_val == "t" && SiteSetting.s3_upload_bucket.blank?
   end
 
+  def validate_secure_images(new_val)
+    validate_error :secure_images_requirements if new_val == "t" && SiteSetting.login_required.blank?
```

Comment:
These two lines needn't be `blank?` because the site setting is typed as a boolean and will never be `nil` right? Could be `SiteSetting.login_required?` etc

Reply:
I switched them to `!SiteSetting.login_required`, but initially I used `blank?` because that was the convention for other validations on the same file. I'm guess they are equivalent.

Reply:
`.blank?` is better for strings or things that might be objects or `nil`. In this case it's a boolean so I think it's better with `!SiteSetting.login_required?`